Source/Core/DolphinWX/FrameTools.cpp

```diff
@@ -862,11 +848,31 @@ void CFrame::DoStop()
     if (NetPlayDialog::GetNetPlayClient())
       NetPlayDialog::GetNetPlayClient()->Stop();
 
+    if (!m_tried_graceful_shutdown && TriggerSTMPowerEvent())
+    {
+      m_tried_graceful_shutdown = true;
+      return;
+    }
     Core::Stop();
     UpdateGUI();
   }
 }
 
+bool CFrame::TriggerSTMPowerEvent()
+{
+  const auto& stm = WII_IPC_HLE_Interface::GetDeviceByName("/dev/stm/eventhook");
```

Comment:
I actually just realized this, but taking this by reference might not be a good idea, since [GetDeviceByName](https://github.com/dolphin-emu/dolphin/blob/a3d2dead76f9fe46163a20f6b7cc6ac7b2b66f29/Source/Core/Core/IPC_HLE/WII_IPC_HLE.cpp#L261) returns by value. Feel free to ignore this though, since it was already present in the original code.